File: Daily Challanges/2026-07-July/2026-07-03.js
function migrateRecord(defaultRecord, record) {
  let result = { ...record };

  for (let key in defaultRecord) {
    if (!(key in result)) {
      result[key] = defaultRecord[key];
    }
  }

  return result;
}

console.log(migrateRecord({ username: "", posts: 0 }, { verified: true }));
console.log(
  migrateRecord({ username: "", posts: 0 }, { username: "camper", posts: 5 }),
);
console.log(
  migrateRecord(
    { username: "", posts: 0, verified: false },
    { username: "camper" },
  ),
);
console.log(
  migrateRecord(
    { username: "", posts: 0 },
    { username: "camper", role: "admin" },
  ),
);
console.log(
  migrateRecord(
    {
      username: "",
      email: "",
      posts: 0,
      verified: false,
      role: "user",
      banned: false,
    },
    {
      username: "camper",
      email: "camper@freecodecamp.org",
      role: "admin",
    },
  ),
);

File: Daily Challanges/2026-07-July/2026-07-03.py
def migrate_record(default_record, record):
    result = record.copy()

    for key, value in default_record.items():
        if key not in result:
            result[key] = value

    return result


print(migrate_record({"username": "", "posts": 0}, {"verified": True}))
print(migrate_record({"username": "", "posts": 0}, {"username": "camper", "posts": 5}))
print(
    migrate_record(
        {"username": "", "posts": 0, "verified": False}, {"username": "camper"}
    )
)
print(
    migrate_record(
        {"username": "", "posts": 0}, {"username": "camper", "role": "admin"}
    )
)
print(
    migrate_record(
        {
            "username": "",
            "email": "",
            "posts": 0,
            "verified": False,
            "role": "user",
            "banned": False,
        },
        {"username": "camper", "email": "camper@freecodecamp.org", "role": "admin"},
    )
)
